DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/tb_utils.py

```diff
@@ -584,3 +585,13 @@ async def tb_get_qualified_ports(device, ports, speed, duplex, required_ports=2)
     err_msg = f"Need {required_ports} ports with the same speed of {speed} and duplex {duplex}"
     assert len(speed_ports) >= required_ports, err_msg
     return speed_ports
+
+
+async def tb_utils_generate_rate(device, port):
+    """
+    Return random rate not bigger than current port speed
+    """
+    out = await Ethtool.show(input_data=[{device.host_name: [{"devname": port}]}], parse_output=True)
+    port_speed = int(out[0][device.host_name]['parsed_output']["speed"][:-4]) * 1_000_000
+    rate = random.randrange(port_speed * 0.05, port_speed * 0.8, 1000)
```

Comment:
Why 0.8?

Reply:
No particular reason 
I chose the limit between 5% and 80% of the port max speed